webpack.config.js

```diff
@@ -105,6 +106,11 @@ const config = {
 		],
 	},
 	plugins: [
+		new DefinePlugin( {
```

Comment:
It surprises me that we don't already have a `DefinePlugin` plugin, and makes me worry that in places where we are expecting `NODE_ENV` substitution to incur dead code removal, that it's not actually happening. I wonder if this was changed at some point...

Reply:
Hmmm #755

Reply:
Hmmm! https://github.com/WordPress/gutenberg/pull/5267/files#diff-11e9f7f953edc64ba14b0cc350ae7b9dL216 @youknowriad

Reply:
@aduth I think it might have been intentional, it looks as though it happens automatically since webpack 4 whenever `mode` is specified—probably as part of their transition to requiring less config:
https://webpack.js.org/guides/production/#specify-the-mode

> Since webpack v4, specifying mode automatically configures DefinePlugin for you

I wonder whether that's still the case when `DefinePlugin` is also used manually 🤔

Reply:
Aha! I was looking for something exactly like that, as I'd been curious if the reason for removing it was that it was automated in such a way. Nice find.

I checked to see that it was folded correctly in what we distribute in core currently:

https://github.com/WordPress/WordPress/blob/c4505be5343f88f0dd5a0c51ab7271f1aaf46c05/wp-includes/js/dist/blocks.js#L8133